src/entities/ether.ts

```diff
@@ -1,27 +1,30 @@
-import invariant from 'tiny-invariant'
 import { Currency } from './currency'
 import { NativeCurrency } from './nativeCurrency'
 import { Token } from './token'
-import { WETH9 } from './weth9'
 
 /**
  * Ether is the main usage of a 'native' currency, i.e. for Ethereum mainnet and all testnets
  */
 export class Ether extends NativeCurrency {
-  protected constructor(chainId: number) {
+  private _wrappedEtherAddress: string
+  private _wrappedEther: Token | undefined
+
+  protected constructor(chainId: number, address: string) {
     super(chainId, 18, 'ETH', 'Ether')
+    this._wrappedEtherAddress = address
```

Comment:
actually..sorry one more note, we probably want to just construct the token in the constructor, not cache it, because we want to validate the weth address, and not store the wrapped ether address. you can just make wrapped a readonly that you set in the constructor